pkg/client/observatorium/client_test.go

```diff
@@ -0,0 +1,358 @@
+package observatorium
+
+import (
+	"fmt"
+	"net/http"
+	"testing"
+
+	. "github.com/onsi/gomega"
+	pAPI "github.com/prometheus/client_golang/api"
+	pV1 "github.com/prometheus/client_golang/api/prometheus/v1"
+)
+
+var (
+	observabilityConfiguration = NewObservabilityConfigurationConfig()
+	configuration              = Configuration{
+		Cookie:   observabilityConfiguration.Cookie,
+		Timeout:  observabilityConfiguration.Timeout,
+		Debug:    observabilityConfiguration.Debug,
+		Insecure: observabilityConfiguration.Insecure,
+		AuthType: "test",
+		BaseURL:  "",
+	}
+	testValue = "test"
+)
+
+func Test_NewObservatoriumClient(t *testing.T) {
+	type args struct {
+		c *ObservabilityConfiguration
+	}
+
+	tests := []struct {
+		name     string
+		modifyFn func(config *ObservabilityConfiguration)
+		args     args
+		wantErr  bool
+	}{
+		{
+			name: "should not return an error when providing default ObservabilityConfiguration",
+			args: args{
+				c: NewObservabilityConfigurationConfig(),
+			},
+			wantErr: false,
+		},
+		{
+			name: "should not return an error when providing default ObservabilityConfiguration with AuthType = AuthTypeSso",
+			args: args{
+				c: NewObservabilityConfigurationConfig(),
+			},
+			modifyFn: func(config *ObservabilityConfiguration) {
+				config.AuthType = AuthTypeSso
+			},
+			wantErr: false,
+		},
+		{
+			name: "should not return an error when providing default ObservabilityConfiguration with mock enabled",
+			args: args{
+				c: NewObservabilityConfigurationConfig(),
+			},
+			modifyFn: func(config *ObservabilityConfiguration) {
+				config.EnableMock = true
+			},
+			wantErr: false,
+		},
+	}
+	RegisterTestingT(t)
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			config := tt.args.c
+			if tt.modifyFn != nil {
+				tt.modifyFn(config)
+			}
+			_, err := NewObservatoriumClient(config)
+			Expect(err != nil).To(Equal(tt.wantErr))
+		})
+	}
+}
+
+func Test_NewClient(t *testing.T) {
+	type args struct {
+		c *Configuration
+	}
+
+	tests := []struct {
+		name     string
+		args     args
+		modifyFn func(config *Configuration)
+		wantErr  bool
+	}{
+		// only one test case required. the rest is covered in the Test_NewObservatoriumClient
+		{
+			name: "should return an error when creating a client with invalid url",
+			args: args{
+				c: &configuration,
+			},
+			modifyFn: func(config *Configuration) {
+				config.BaseURL = ":::" // invalid url
+			},
+			wantErr: true,
+		},
+	}
+	RegisterTestingT(t)
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			config := tt.args.c
+			if tt.modifyFn != nil {
+				tt.modifyFn(config)
+			}
+			_, err := NewClient(config)
+			Expect(err != nil).To(Equal(tt.wantErr))
```

Comment:
if as described in the comment no other test cases are needed, we an remove `wantErr` and change this to:
```suggestion
			Expect(err).To(HaveOccurred())
```